src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.cs

```diff
@@ -20,5 +24,55 @@ internal static void VerifyValidPath(string path, string argName)
                 throw new ArgumentException(SR.Argument_InvalidPathChars, argName);
             }
         }
+
+        internal static bool Copy(string sourcePath, string destinationPath, bool recursive,
+            bool skipExistingFiles, CancellationToken cancellationToken)
+        {
+            var searchOption = recursive ? SearchOption.AllDirectories : SearchOption.TopDirectoryOnly;
+            IEnumerable<string> directoryEnumeration;
+
+            try
+            {
+                directoryEnumeration = Directory.EnumerateDirectories(sourcePath, "*", searchOption);
+            }
+            catch (IOException)
+            {
+                // Return false if Enumeration is not possible
+                return false;
+            }
+
+            foreach (string enumeratedDirectory in directoryEnumeration)
+            {
+                string newDirectoryPath = enumeratedDirectory.Replace(sourcePath, destinationPath);
+                Directory.CreateDirectory(newDirectoryPath);
+
+                cancellationToken.ThrowIfCancellationRequested();
+            }
+
+            foreach (string enumeratedFile in Directory.GetFiles(sourcePath, "*.*", searchOption))
+            {
+                if (skipExistingFiles && File.Exists(enumeratedFile))
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+                    continue;
+                }
+
+                try
+                {
+                    CopyFile(enumeratedFile, enumeratedFile.Replace(sourcePath, destinationPath), true);
+                }
+                catch (IOException ex)
+                {
+                    // Return false for read failures
+                    return ex.HResult == Interop.Errors.ERROR_ACCESS_DENIED ? false :
+                        // Otherwise rethrow
+                        throw ex;
```

Comment:
why not just have it do ``throw`` to preserve the original exception (to not wrap it in a new exception)

Reply:
See my other comment, I believe it should continue with next file instead of immediately return.

Reply:
@adamsitnik Could you confirm if the part is a proper implementation?